exe/Main.hs

```diff
@@ -525,8 +539,7 @@ liftCSE subrecord name body expr =
         Expr.App f a ->
           Expr.App <$> go f v <*> go a v
 
-        Expr.Let n t b e ->
-          Expr.Let n t <$> go b v <*> go e ( shiftName n v )
+        Expr.Let bs e -> uncurry Expr.Let <$> goLet v bs e
```

Comment:
I like @jneira's implementation in #143 better:

```
        Expr.Let bs e ->
          Expr.Let <$> traverse go' bs <*> go e shifted
            where go' (Expr.Binding n t b) = Expr.Binding n t <$> go b v
                  shifted = foldr (shiftName . Expr.variable) v bs 
```

Then there's no need for `goLet`.